cylc/flow/config.py

```diff
@@ -551,21 +553,21 @@ def __init__(
 
         self.configure_suite_state_polling_tasks()
 
-        # Warn or abort (if --strict) if naked dummy tasks (no runtime
+        self._check_task_event_handlers()
+        self._check_special_tasks()  # adds to self.naked_tasks
```

Comment:
This is the old `check_tasks` method split into two parts and moved up slightly earlier in the logic to allow unification of naked task checking.